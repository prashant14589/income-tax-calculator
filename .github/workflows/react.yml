name: React CI

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - name: Checkout
      uses: actions/checkout@v2
    - name: Install dependencies
      run: npm install
    - name: Run jest
      run: ./node_modules/.bin/jest src --passWithNoTests
    - name: Cypress run
      uses: cypress-io/github-action@v2
      if: always()
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      with:
        build: npm run build --if-present
        start: npm start
        wait-on: 'http://localhost:3000'
