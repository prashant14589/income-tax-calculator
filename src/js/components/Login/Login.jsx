import React,{useEffect, useState} from "react";
import {TextInput} from "../Input/TextInput";
const Login = ({handleLoginClick}) => {
    const [username, setUserName] = useState("");
    const [password, setPassword] = useState("");
    const [isValidUser, setIsValidUser] = useState(false); 
    
    useEffect(() => {        
        if(username == "A" && password == "B"){
            setIsValidUser(true);
        }
        else{
            setIsValidUser(false);
        }        
    },[username, password]);
return (
    <>
      <h1>Login</h1>
      <TextInput id="username" value={username} onChange={(e) => setUserName(e.target.value)}>Username:</TextInput> 
      <TextInput id="password" value={password} onChange={(e) => setPassword(e.target.value)}>Password:</TextInput> 
      <button disabled = {!isValidUser} onClick= {handleLoginClick}>Login</button>
    </>
  );
}
  export default Login;