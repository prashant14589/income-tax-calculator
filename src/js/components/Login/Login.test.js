import React from 'react';
import {fireEvent, render, screen } from '@testing-library/react';
import  Login  from './Login';
describe('<Login />', () => {
  const handleLoginClick = jest.fn();
    it('should render successfully', () => {
      const { container } = render(<Login handleLoginClick = {handleLoginClick}/>);
      expect(container).toMatchSnapshot();
    });  
    beforeEach( () => {
      render(<Login handleLoginClick = {handleLoginClick}/>);
    });
    it('should render username field successfully', () => {
    
        expect(screen.getByText('Username:')).toBeInTheDocument();
      });
      it('should render password field successfully', () => {
        
        expect(screen.getByText('Password:')).toBeInTheDocument();
      });
});