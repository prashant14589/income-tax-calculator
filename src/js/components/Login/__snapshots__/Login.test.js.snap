// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render successfully 1`] = `
.c0 {
  padding: 0.5em;
  display: inline-block;
  width: 30%;
  margin: 0.5em;
  color: palevioletred;
  background: papayawhip;
  border: none;
  border-radius: 3px;
}

<div>
  <h1>
    Login
  </h1>
  <div>
    <label
      for="username"
    >
      Username:
    </label>
    <input
      class="c0"
      id="username"
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      for="password"
    >
      Password:
    </label>
    <input
      class="c0"
      id="password"
      type="text"
      value=""
    />
  </div>
  <button
    disabled=""
  >
    Login
  </button>
</div>
`;
